src/world/area_end/end_00/DF6A20.c

```diff
@@ -66,6 +66,27 @@ ApiStatus func_802427D8_DF91D8(ScriptInstance* script, s32 isInitialCall) {
     return ApiStatus_DONE2;
 }
 
-INCLUDE_ASM(s32, "world/area_end/end_00/DF6A20", func_80242830_DF9230);
+// setNpcShadowScale
+ApiStatus func_80242830_DF9230(ScriptInstance* script, s32 isInitialCall) {
+    s32 npcID;
+    f32 newShadowScale;
 
-INCLUDE_ASM(s32, "world/area_end/end_00/DF6A20", func_80242898_DF9298);
+    Bytecode* args = script->ptrReadPos;
+    npcID = get_variable(script, *args++);
+    newShadowScale = get_float_variable(script, *args);
+
+    resolve_npc(script, npcID)->shadowScale = newShadowScale;
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_80242898_DF9298(ScriptInstance* script, s32 isInitialCall) {
+    Npc* npc;
+    Bytecode* args = script->ptrReadPos;
+    if (isInitialCall != 0) {
+        script->functionTemp[1] = get_npc_unsafe(get_variable(script, *args));
+    }
+    npc = (Npc*)script->functionTemp[1];
+    npc->pos.x += 0.6666667f;
```

Comment:
IMO its cleaner to type-alias a pointer to script->functionTemp[1] (and if it was an f32 instead of an Npc* it'd be required to match). Try something like this:
```suggestion
    Bytecode* args = script->ptrReadPos;
    Npc** npc = &script->functionTemp[1];
 
    if (isInitialCall) {
        *npc = get_npc_unsafe(get_variable(script, *args));
    }

    (*npc)->pos.x += 0.6666667f;
```
Also note that isInitialCall is a bool, so we tend to just write `if (isInitialCall)` (`!= 0` is implicit).